File: 정민지/java_algo/src/sea/SEA_4835.java
package sea;

import java.io.*;
import java.util.*;
import java.util.stream.IntStream;

public class SEA_4835 {

	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

		int testCase = Integer.parseInt(br.readLine());
		for (int tc = 0; tc < testCase; tc++) {
			StringTokenizer st = new StringTokenizer(br.readLine());
			int n = Integer.parseInt(st.nextToken());
			int m = Integer.parseInt(st.nextToken());
			int[] arr = Arrays.stream(br.readLine().split(" ")).mapToInt(Integer::parseInt).toArray();

			int maxVal = Integer.MIN_VALUE;
			int minVal = Integer.MAX_VALUE;
			for (int i = 0; i < n - m + 1; i++) {
				int sum = IntStream.of(Arrays.copyOfRange(arr, i, i+m)).sum();
				if (sum > maxVal) maxVal = sum;
				if (sum < minVal) minVal = sum;
			}
			bw.write(String.format("#%d %d%n", tc+1, maxVal-minVal));
		}
		bw.flush();
		bw.close();
	}
}

File: 정민지/python/sea/SEA_4835.py
import sys

input = sys.stdin.readline

for tc in range(int(input())):
    n, m = map(int, input().split())
    lst = list(map(int, input().split()))

    minVal = sys.maxsize
    maxVal = -sys.maxsize - 1

    for i in range(n - m + 1):
        sumVal = sum(lst[i:i + m])

        if sumVal > maxVal:
            maxVal = sumVal
        if sumVal < minVal:
            minVal = sumVal

    print(f'#{tc + 1} {maxVal - minVal}')
